src/js/components/pages/layout-text-editor.jsx

```diff
@@ -371,14 +377,16 @@ var SceneMonacoTextEditor = React.createClass({
         return compareNewPropsAndCurrentEditorCopy;
 
     },
-    componentWillReceiveProps: function (nextProps) {
-        /* HubSendActions.loadScene(nextProps._id);*/
-    },
+
     componentDidUpdate: function (previousProps, previousState) {
 
         try {
-            if (!_.isEqual(this.state.scene, previousState.scene))
+            var sceneChangeShouldUpdate = !_.isEqual(this.state.scene, previousState.scene);
+            var emptyEditorShouldUpdate = this.getMonacoEditorVersionOfScene() === null;
+            if (sceneChangeShouldUpdate || emptyEditorShouldUpdate) {
+                // APEP after updating, make sure we update the monaco editor
                 this.refs.monaco.editor.setValue(this.getSceneString());
+            }
         } catch (e) {
             console.log("Error with bad json: ", e);
```

Comment:
Ideally we should update the try catch to catch either JSON errors, or null pointers to the monaco.editor DOM